frb_codegen/src/generator/rust/ty_general_list.rs

```diff
@@ -17,22 +17,36 @@ impl TypeGeneralListGenerator<'_> {
 
 impl TypeRustGeneratorTrait for TypeGeneralListGenerator<'_> {
     fn wire2api_body(&self) -> Option<String> {
-        Some(TypeGeneralListGenerator::WIRE2API_BODY.to_string())
+        Some(match &self.ir {
+            IrTypeGeneralList::GeneralList { .. } => TypeGeneralListGenerator::WIRE2API_BODY.into(),
+            IrTypeGeneralList::GeneralArray { len, .. } => format!(
```

Comment:
Looks like the generated code of `GeneralArray` works very similarly as `GeneralList`. Is it possible we do not duplicate this code string here, but instead merge it with `TypeGeneralListGenerator::WIRE2API_BODY`?